### DISCUSSION TITLE
VRAM usage at different resolutions

### DISCUSSION THREAD

**Discussion by HenkDz** [hlky/stable-diffusion#80] [General]
What is the largest size I can get with my RTX 3090 24GBvRAM?

![image](https://user-images.githubusercontent.com/17301927/187094697-8d33d149-d618-445b-a739-fa77f205ec7f.png)

**Comment by hlky** [comment]
VRAM usage at different resolutions is something that needs documenting, larger resolutions require more vram, and from my previous tests it's roughly linear, for example, an image that was 512x512 used around 6gb, and an image that was 512x1024 (double the size), used just under 12gb.

Accidentally lost the resolution you were trying when I changed the title but I think it was 768x832 which is roughly 22% larger than 512x1024 so I would expect it to use roughly 22% more vram

So then, I'm not sure why you are running out of vram considering you have 24gb, are you running with full precision? full precision increases vram usage, if you're not sure, that means you haven't specifically enabled full precision, half precision is default

Do you happen to have other things open using vram, games, videos, even web browser uses vram

Btw, vram usage in the webui is subject to change because we use other models as well i.e. gfpgan and esrgan, you can always try removing the model files for those to stop them being loaded

**Reply by HenkDz** [reply]
as you can see in the console message it says I have 9.57GB free yet it exited. I have nothing running but chrome browser and the webui tab. I have got 32GB of ram and Rayzen 5950 CPU.

**Reply by HenkDz** [reply]
I'm doing 832x640 fine but seems this is as far as I can go

![image](https://user-images.githubusercontent.com/17301927/187095893-ec07503c-d1d4-495e-9826-9a0ce9206f98.png)

**Reply by HenkDz** [reply]
Go the error again when trying to upscale using GoBIG. Is there a way I can allow Python to use the full 24GB?

![image](https://user-images.githubusercontent.com/17301927/187096072-e8011684-f97f-4434-9543-9a894724b226.png)

**Reply by hlky** [reply]
@HenkDz gobig was temporarily removed so no support for that specifically right now, plus it means you're on an old version

**Reply by hlky** [reply]
> as you can see in the console message it says I have 9.57GB free yet it exited. I have nothing running but chrome browser and the webui tab. I have got 32GB of ram and Rayzen 5950 CPU.

The numbers in the error message unrelated and don't actually mean you have 9.57GB free, if you get an error like that all it means is you have ran out of vram, and the first number is by how much, ignore the rest

**Reply by hlky** [reply]
Also, as already asked, are you running with full precision? full precision increases vram usage, if you're not sure, that means you haven't specifically enabled full precision, half precision is default

**Reply by HenkDz** [reply]
I didn't specify anything, so it is running on half precision by default.

**Reply by hlky** [reply]
we use other models as well i.e. gfpgan and esrgan, you can always try removing the model files for those to stop them being loaded to reduce vram

if a large image is what you're after consider just using the esrgan tab on any images you want upscaled, also usable on generation but generally you won't want every image generated to be upscaled

**Reply by HenkDz** [reply]
I thought the double image was fixed??? Disabling save grid as a workaround.

![image](https://user-images.githubusercontent.com/17301927/187097350-ce6d54b1-c1f0-481a-9637-1533cfff738a.png)

**Reply by hlky** [reply]
It should be, update

**Comment by HenkDz** [comment]
Still struggling with memory management. Very low resolution. 512x640. 60 Steps and 10 images (separate not a batch) 

![image](https://user-images.githubusercontent.com/17301927/188320473-b6ffcc89-da4c-4f9c-9db0-5b324696f227.png)
